### PR TITLE
Allow array of user IDs for Notify operation

### LINKED ISSUE #14273: Allow array of IDs passed to Notify operation in Flows
### Describe the Bug

I'm trying to send `Notifications` to users that are subscribers.

In my test I have two simple collections; `Pages` and `Subscriptions`. The second collection has one column named `user_id`. 

In my Flow I (1) trigger on created pages, (2) get all subscribing users and (3) send notifications. If I hard code a single user id in the user field it works, but if I use the Flow object it doens't create any notification at all.

Maybe it's because that object is an array with multiple user id's? But in the docs example it seems to be possible with emails... What is the correct syntax?

Skärmavbild 2022-07-04 kl  14 11 28
Skärmavbild 2022-07-04 kl  14 11 41

### To Reproduce

Se above

### Errors Shown

_No response_

### What version of Directus are you using?

9.13.0

### What version of Node.js are you using?

17.4.0

### What database are you using?

PG 13

### What browser are you using?

Chrome

### How are you deploying Directus?

Locally with npm

### PR DESCRIPTION
**Pull request by azrikahar**
## Description

Closes #14273

`User` is now a tags interface to accept an array, similar to the `To` field for recipient emails in Email oepration:

chrome_emjWn5Wihn

## Type of Change

- [ ] Bugfix
- [x] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated

### CODE REVIEW

**Comment by danieljwestman** [comment]
[azrikahar](https://github.com/azrikahar), Will this PR solve the issue I've posted? https://github.com/directus/directus/issues/14273

**Comment by azrikahar** [comment]
> Will this PR solve the issue I've posted? #14273

Technically only partially in terms of the use case you had in #14273, mainly because Read Data returns array of objects instead of array of IDs. I've shared more detail about it here: https://github.com/directus/directus/issues/14273#issuecomment-1196495124 

The additional transform capability would make it so that we can re-use it anywhere else (such as an array of user emails in Email operation), so it's best not to couple it to this PR as it'd be a separate new feature. 👍

**Review by rijkvanzanten — APPROVED** [review]

File: app/src/operations/notification/index.ts [hunk 1]
```diff
@@ -22,10 +22,10 @@ export default defineOperationApp({
 			type: 'string',
```
Comment (rijkvanzanten) [inline app/src/operations/notification/index.ts:22]: Wondering if we should update this to `json`. Thoughts @azrikahar?

Comment (azrikahar) [reply app/src/operations/notification/index.ts:22]: 100%! Purely an oversight on my part. #14021 updated a number of them from `string` to `json`, so it definitely makes sense to follow suit here.

Comment (azrikahar) [reply app/src/operations/notification/index.ts:22]:
```suggestion
			type: 'json',
```

File: app/src/operations/notification/index.ts [hunk 2]
```diff
@@ -19,13 +19,13 @@ export default defineOperationApp({
 		{
 			field: 'recipient',
 			name: '$t:operations.notification.recipient',
-			type: 'string',
+			type: 'json',
```
Comment (rijkvanzanten) [inline app/src/operations/notification/index.ts:22]: And now for the fun question: will the `tags` interface break when `json` is used but an (old) single string value is passed? 😄

Comment (azrikahar) [reply app/src/operations/notification/index.ts:22]: Just tested it and it doesn't seem to break as the fix for tags interface in the raw editor PR does prevent it from breaking 😄

That being said, you did make me fully realize the change to json was never included with tags... 🤦‍♂️ 

Currently all the tags interface fields across all Flow operations are actually using `csv`, with one exception being `To` in Email operation which is still using `string`:

https://github.com/directus/directus/blob/080f729079e5e998e8d651b06dcdbfb4fd6d8c5d/app/src/operations/mail/index.ts#L24-L29

Should we make them all csv for uniformity instead?

Comment (rijkvanzanten) [reply app/src/operations/notification/index.ts:22]: > Currently all the tags interface fields across all Flow operations are actually using csv, with one exception being To in Email operation which is still using string:

Oh! In that case, lets use `csv` everywhere yeah 👍🏻

Comment (azrikahar) [reply app/src/operations/notification/index.ts]: Done 👍

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**